src/server/console/default_configuration.cpp

```diff
@@ -105,8 +106,11 @@ std::shared_ptr<mir::ConsoleServices> mir::DefaultServerConfiguration::the_conso
                 {
                     try
                     {
+                        auto session_lock_instance = the_session_lock();
                         auto const vt_services = std::make_shared<mir::LogindConsoleServices>(
-                            std::dynamic_pointer_cast<mir::GLibMainLoop>(the_main_loop()));
+                            std::dynamic_pointer_cast<mir::GLibMainLoop>(the_main_loop()),
+                            session_lock_instance);
+                        session_lock_instance->register_interest(vt_services);
```

Comment:
It looks a bit odd to `register_interest()` here, and not in the constructor. I'd probably prefer the "named constructor idiom" - make the actual constructor private and add a static method to `LogindConsoleServices`:

```c++
auto LogindConsoleServices::create(std::shared_ptr<MainLoop> ml, std::shared_ptr<scene::SessionLock> const& session_lock)
-> std::shared_ptr<mir::LogindConsoleServices>
{
    auto cs = std::make_shared<mir::LogindConsoleServices>(std::move(the_main_loop()), session_lock);
    session_lock->register_interest(vt_services);
    return cs;
}
```

Reply:
Oh! You actually need `GLibMainLoop` - I'd move the necessary casts into the constructor/named constructor

Reply:
Also, i'd like to leave the cast in `the_console_services()`, as you technically can't know if the MainLoop coming in has the right type